clang/lib/CIR/CodeGen/CIRGenDecl.cpp

```diff
@@ -59,7 +55,10 @@ CIRGenFunction::buildAutoVarAlloca(const VarDecl &D) {
 
   Address address = Address::invalid();
   Address allocaAddr = Address::invalid();
-  Address openMPLocalAddr = Address::invalid();
+  Address openMPLocalAddr =
+      getCIRGenModule().getOpenMPRuntime().getAddressOfLocalVariable(*this, &D);
+  if (getLangOpts().OpenMPIsTargetDevice)
+    assert(UnimplementedFeature::openMPTarget());
```

Comment:
Not sure I understand this one. The way this is written, it translates to `assert(!getLangOpts().OpenMPIsTargetDevice && "NYI")`. If this is what you want we don't need an implemented feature check here and asserting directly on the option is better. If you are ignoring the language option for now, then it should be `assert(!UnimplementedFeature::openMPTarget());`